esmvaltool/cmorizers/data/cmorizer.py

```diff
@@ -0,0 +1,527 @@
+        logger.info("CMORizing dataset %s using Python script %s", dataset,
+                    module.__file__)
+        cmor_cfg = read_cmor_config(dataset)
+        module.cmorization(in_dir, out_dir, cmor_cfg, self.config, start, end)
+        logger.info('CMORization of dataset %s finished!')
+        return True
+
+
+class DataCommand():
+    """Download and format data to use with ESMValTool."""
+    def __init__(self):
+        datasets_file = os.path.join(os.path.dirname(__file__), 'datasets.yml')
+        with open(datasets_file) as data:
+            self._info = yaml.safe_load(data)
+        self.formatter = Formatter(self._info)
+
+    def list(self):
+        """List all supported datasets."""
+        print()
+        print(f'| {"Dataset name":30} | Tier | Auto-download | Last access |')
+        print('-' * 71)
+        for dataset, dataset_info in self._info['datasets'].items():
+            date = datetime.datetime.strptime(str(dataset_info['last_access']),
+                                              "%Y-%m-%d")
+            print(f"| {dataset:30} | {dataset_info['tier']:4} "
+                  f"| {self.formatter.has_downloader(dataset):13} "
+                  f"|  {date.strftime('%Y-%m-%d')} |")
+        print('-' * 71)
+
+    def info(self, dataset):
+        """Show detailed info about a specific dataset.
+
+        Parameters
+        ----------
+        dataset : str
+            dataset to show
+        """
+        dataset_info = self._info['datasets'][dataset]
+        print(dataset)
+        print()
+        print(f"Tier: {dataset_info['tier']}")
+        print(f"Source: {dataset_info['source']}")
+        print(f"Automatic download: {self.formatter.has_downloader(dataset)}")
+        print("")
+        print(dataset_info['info'])
+
+    def download(self,
+                 datasets,
+                 config_file=None,
+                 start=None,
+                 end=None,
+                 overwrite=False,
+                 **kwargs):
+        """Download datasets.
+
+        Parameters
+        ----------
+        datasets : list(str), optional
+            List of datasets to format, by default None
+        config_file : str, optional
+            Path to ESMValTool's config user file, by default None
+        start : str, optional
+            Start of the interval to process, by default None. Valid formats
+            are YYYY, YYYYMM and YYYYMMDD.
+        end : str, optional
+            End of the interval to process, by default None. Valid formats
+            are YYYY, YYYYMM and YYYYMMDD.
+        overwrite : bool, optional
+            If true, download already present data again
+        """
+        start = self._parse_date(start)
+        end = self._parse_date(end)
+
+        self.formatter.start('download', datasets, config_file, kwargs)
+        self.formatter.download(start, end, overwrite)
+
+    def format(self,
+               datasets=None,
```

Comment:
I find this default option too dangerous. When using `esmvaltool data format` and omitting the dataset(s), the tool starts to cmorize all supported datasets. I think it would be better not to make `datasets` optional. If no dataset is passed as argument, the tool should not do anything or crash but not process everything.

Reply:
This was the original behavior, but I agree that now is probably to restrict it a bit more